scrapy/http/request/form.py

```diff
@@ -51,7 +54,10 @@ def from_response(cls, response, formname=None, formid=None, formnumber=0, formd
 
 def _get_form_url(form, url):
     if url is None:
-        return urljoin(form.base_url, form.action)
```

Comment:
this urljoin was pointless because form.action already calls `urljoin(form.base_url, action)`